include/boost/config/compiler/xlcpp_zos.hpp

```diff
@@ -0,0 +1,201 @@
+//  Copyright (c) 2017 Dynatrace
+//
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt
+
+//  See http://www.boost.org for most recent version.
+
+//  Compiler setup for IBM z/OS XL C/C++ compiler.
+
+// Oldest compiler version currently supported is 2.1 (V2R1)
+#if !defined(__IBMCPP__) || !defined(__COMPILER_VER__) || __COMPILER_VER__ < 0x42010000
+#  error "Compiler not supported or configured - please reconfigure"
+#endif
+
+#if __COMPILER_VER__ > 0x42010000
```

Comment:
We don't normally have hard failures just because the compiler has been updated, there may be new features in the new compiler that we don't detect, but that's normally safe anyway.  Please wrap this in a #ifdef BOOST_ASSERT_CONFIG block.  Thanks!